components/exercises/chapter-6-oscar/VectorType/VectorType.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState } from 'react';
+import { Vec } from './Vec'; // Import the Vec class
+
+/**
+ * Represents a square and provides controls to increase or decrease the size.
+ */
+export default function VectorType() {
+  const [vector, setVector] = useState(new Vec(60, 60)); // Initial vector with width and height
+
+  // Event handler for changing the width input
+  const handleWidthChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newWidth = Number(event.target.value);
+    setVector((prevVector) => new Vec(newWidth, prevVector.y));
+  };
+
+  // Event handler for changing the height input
+  const handleHeightChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newHeight = Number(event.target.value);
+    setVector((prevVector) => new Vec(prevVector.x, newHeight));
+  };
+
+  // Event handler for increasing the vector by 10 in both width and height
+  const handlePlus = () => {
+    setVector((prevVector) => prevVector.plus(new Vec(10, 10)));
+  };
+
+  // Event handler for decreasing the vector by 10 in both width and height
+  const handleMinus = () => {
+    setVector((prevVector) => prevVector.minus(new Vec(10, 10)));
+  };
+
+  // Event handler for displaying an alert with the length of the vector
+  const handleLength = () => {
+    alert(`Length: ${vector.length}`);
+  };
+
+  return (
+    <div className='flex flex-col items-center justify-center'>
+      <div
+        className={`w-${vector.x} h-${vector.y} bg-blue-500`}
```

Comment:
This className doesn't do anything since the inline styling in the next line takes precedence over it. 
```diff
- className={`w-${vector.x} h-${vector.y} bg-blue-500`}
+ className="bg-blue-500"
```